app/grandchallenge/notifications/forms.py

```diff
@@ -39,16 +36,17 @@ def clean(self):
             raise ValidationError("You cannot create this subscription!")
 
         if cleaned_data["content_type"] == ContentType.objects.get(
-            app_label="forum", model="forum"
+            app_label="discussion_forums", model="forum"
         ):
-            perm = self.perm_handler.can_read_forum(
-                Forum.objects.filter(id=obj_id).get(), self.user
+            perm = self.user.has_perm(
+                "view_forum", Forum.objects.filter(pk=obj_id).get()
             )
         elif cleaned_data["content_type"] == ContentType.objects.get(
-            app_label="forum_conversation", model="topic"
+            app_label="discussion_forums", model="forumtopic"
         ):
-            perm = self.perm_handler.can_subscribe_to_topic(
-                Topic.objects.filter(id=obj_id).get(), self.user
+            perm = self.user.has_perm(
+                "view_forumtopic",
+                ForumTopic.objects.filter(pk=obj_id).get(),
             )

```

Comment:
It seems strange to me to do a permission check in the form clean method. If I had to implement this now from scratch, I would move the check to the view. That would require constructing the form in the views where it is used and pass it into the context of the template through the view. The template tag that we're currently using could then be removed. That feels like a cleaner solution, but I figured I first get things working, and then optimize. What do you think about this, James?

Reply:
> figured I first get things working, and then optimize

Yes, this! Additionally always separate refactoring from new functionality. Make a separate PR with the refactoring onto main, then rebase your feature.